.github/workflows/validate_bids-examples.yml

```diff
@@ -0,0 +1,144 @@
+name: validate_datasets
+
+on:
+  push:
+    branches: ['master']
+  pull_request:
+    branches: ['**']
+#  create:
+#    branches: [master]
+#    tags: ['**']
+#  schedule:
+#    - cron: "0 4 * * 1"
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [ubuntu-latest] # , macos-latest, windows-latest]
+        bids-validator: [master-deno]
+        python-version: ["3.11"]
+
+    runs-on: ${{ matrix.platform }}
+
+    env:
+      TZ: Europe/Berlin
+      FORCE_COLOR: 1
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: "Apply patches for BIDS-2.0"
+      run: |
+        set -o pipefail
+        tools/bids-2.0/apply_all 2>&1 | tee /tmp/patch.log
+
+    - name: "Show differences after patching"
+      run: |
+        git add .
+        git diff --cached
+
+    - name: "Commit and push patched version online for possible introspection"
+      # Run only on a non-merge commit for the PR
+      if: >
+        github.repository == 'bids-standard/bids-specification' &&
+        github.event.pull_request.head.ref == 'bids-2.0' &&
```

Comment:
might want to extend here to any branch with `bids-2.0` prefix but without `-patched` suffix so we support rendering of diffs etc for feature branches against this branch, such as in

- https://github.com/bids-standard/bids-specification/pull/1809